app/Controller/Board.php

```diff
@@ -247,6 +247,39 @@ public function changeAssignee()
     }
 
     /**
+     * Change a task assignee to me from the task
+     *
+     * @access public
+     */
+    public function assignToMe()
+    {
+        $task = $this->getTask();
+
+        $assignee = $this->user->getByUsername($task['assignee_username']);
```

Comment:
You don't need to do that, the assignee id is already present in the task data (field owner_id)